sdks/java/io/cassandra/src/main/java/org/apache/beam/sdk/io/cassandra/CassandraIO.java

```diff
@@ -0,0 +1,277 @@
+      checkArgument(hosts.isEmpty(), "CassandraIO.read().withHosts(hosts) called with empty "
+          + "hosts list");
+      return builder().setHosts(hosts).build();
+    }
+
+    public Read<T> withPort(int port) {
+      checkArgument(port > 0, "CassandraIO.read().withPort(port) called with invalid port "
+          + "number (" + port + ")");
+      return builder().setPort(port).build();
+    }
+
+    public Read<T> withKeyspace(String keyspace) {
+      checkArgument(keyspace != null, "CassandraIO.read().withKeyspace(keyspace) called with "
+          + "null keyspace");
+      return builder().setKeyspace(keyspace).build();
+    }
+
+    public Read<T> withTable(String table) {
+      checkArgument(table != null, "CassandraIO.read().withTable(table) called with null table");
+      return builder().setTable(table).build();
+    }
+
+    public Read<T> withEntity(Class<T> entity) {
+      checkArgument(entity != null, "CassandraIO.read().withEntity(entity) called with null "
+          + "entity");
+      return builder().setEntity(entity).build();
+    }
+
+    public Read<T> withCoder(Coder<T> coder) {
+      checkArgument(coder != null, "CassandraIO.read().withCoder(coder) called with null coder");
+      return builder().setCoder(coder).build();
+    }
+
+    public Read<T> withCassandraService(CassandraService<T> cassandraService) {
+      checkArgument(cassandraService != null, "CassandraIO.read().withCassandraService(service)"
+          + " called with null service");
+      return builder().setCassandraService(cassandraService).build();
+    }
+
+    @Override
+    public PCollection<T> expand(PBegin input) {
+      return input.apply(org.apache.beam.sdk.io.Read.from(
+          new CassandraSource<T>(this, null)));
+    }
+
+    @Override
+    public void validate(PBegin input) {
+      checkState(hosts() != null || cassandraService() != null,
+          "CassandraIO.read() requires a list of hosts to be set via withHosts(hosts) or a "
+              + "Cassandra service to be set via withCassandraService(service)");
+      checkState(port() != null || cassandraService() != null, "CassandraIO.read() requires a "
+          + "valid port number to be set via withPort(port) or a Cassandra service to be set via "
+          + "withCassandraService(service)");
+      checkState(keyspace() != null, "CassandraIO.read() requires a keyspace to be set via "
+          + "withKeyspace(keyspace)");
+      checkState(table() != null, "CassandraIO.read() requires a table to be set via "
+          + "withTable(table)");
+      checkState(entity() != null, "CassandraIO.read() requires an entity to be set via "
+          + "withEntity(entity)");
+      checkState(coder() != null, "CassandraIO.read() requires a coder to be set via "
+          + "withCoder(coder)");
+    }
+
+    @Override
+    public void populateDisplayData(DisplayData.Builder builder) {
```

Comment:
note that this display data won't show up in certain runners since it's a composite ptransform - it's suggested that instead you put this on the source.

@bjchambers do we have this well documented somewhere?